src/shared/utils/utils.datasource.ts

```diff
@@ -39,7 +39,7 @@ export class DataSourceFetcher {
    * @param type - The type of data source to retrieve ('prometheus' or loki)
    * @returns Array of healthy data sources
    */
-  public async getHealthyDataSources(type: DataSourceType): Promise<DataSource[]> {
+  public async getHealthyDataSources(type: DataSourceType, limit = 25): Promise<DataSource[]> {
```

Comment:
I'm open to feedback about `limit = 15`. 15 is certainly a lot more than the 5 Loki data sources that we allow in the data sources drop-down for Related Logs. I think that this value should be higher than 5 to account for the potential for (a) unhealthy datasources and (b) data sources that take longer than 3s to respond as healthy, but I'm not sure how much higher.